bin/problem.py

```diff
@@ -916,38 +918,104 @@ def validate_data(problem, mode: validate.Mode, constraints: dict | bool | None
+            else (
+                f"Collecting {mode} constraints" if constraints else f"{mode} validation"
+            ).capitalize()
+        )
+
+        testcases = problem.testcases(mode=mode)
+        return problem._validate_data(mode, constraints, action, testcases)
+
+    def validate_invalid_extra_data(p) -> bool:
+        base_path = p.tmpdir / "invalid_data"
+        # find at least one (valid?) testcase to modify
+        test_paths = list(glob(p.path, "data/sample/**/*.in"))
+        test_paths += list(glob(p.path, "data/secret/**/*.in"))
+        test_paths = [path for path in test_paths if path.with_suffix(".ans").exists()]
+        test_paths.sort()
+        test_path = test_paths[0] if test_paths else None
+        if test_path:
+            test_case = test_path.relative_to(p.path / "data").with_suffix("")
+            log(f"Generating invalid testcases based on: {test_case}")
+        debug(f"writing generated invalid testcases to: {base_path}")
+
+        # validator, dir, read, write,  copy, allow_whitespace_changes
+        validators: list[tuple[type[validate.AnyValidator], str, str, str, list[str], bool]] = [
+            (validate.InputValidator, "invalid_inputs", ".in", ".in", [], False),
+            (validate.AnswerValidator, "invalid_answers", ".ans", ".ans", [".in"], False),
+            (validate.OutputValidator, "invalid_outputs", ".ans", ".out", [".in", ".ans"], True),
+        ]
+
+        testcases = []
+        for cls, directory, read, write, copy, allow_whitespace_changes in validators:
+            if (p.interactive or p.multipass) and cls != validate.InputValidator:
+                continue
+            if not p.validators(cls, strict=True, print_warn=False):
+                continue
+            if test_path is None and copy:
+                continue
+
+            for name, data, only_whitespace_change in validator_tests.GENERATORS:
+                if allow_whitespace_changes and only_whitespace_change:
+                    continue
+
+                if isinstance(data, str):
+                    content = data
+                elif test_path:
+                    valid = test_path.with_suffix(read).read_text()
+                    generated = data(valid)
+                    if generated is None:
+                        continue
+                    content = generated
+
+                short_path = Path(directory) / name
+                full_path = base_path / short_path / "testcase.in"
+                full_path.parent.mkdir(parents=True, exist_ok=True)
+
+                for ext in copy:
+                    assert test_path is not None
+                    shutil.copy(test_path.with_suffix(ext), full_path.with_suffix(ext))
+                full_path.with_suffix(write).write_text(content)
+
+                testcases.append(testcase.Testcase(p, full_path, short_path=short_path))
+
+        return p._validate_data(validate.Mode.INVALID, None, "Generic Invalidation", testcases)
```

Comment:
Currently, `doc/validation.md` only talks about Input/Answer/Output (in)validation, I think it would be good to add a section about this new "Generic Invalidation" as well :slightly_smiling_face: